.github/workflows/e2e-matrix.yaml

```diff
@@ -51,6 +53,7 @@ jobs:
   e2e:
     strategy:
       fail-fast: false
+      max-parallel: ${{ inputs.parallelism || 100 }}
```

Comment:
Have you considered using `concurrency`? https://docs.github.com/en/actions/using-jobs/using-concurrency